### PR TITLE
chore: framer-motion v11

### LINKED ISSUE #6379: Update `framer-motion` to latest compatible

### PR DESCRIPTION
**Pull request by AlexanderMelox**
Closes #6379 

This PR updates framer motion from version 6 -> 11

#### What did you change?
Update framer-motion package via `yarn up framer-motion@11`

#### How did you test and verify your work?
[Read the migration guide for v11](https://motion.dev/docs/upgrade-guide)
Tested components that utilize framer-motion

### CODE REVIEW

**Comment by AlexanderMelox** [comment]
Screenshot 2024-11-20 at 9 58 51 AM

Not too sure about this build error

File: packages/ibm-products/package.json [hunk 1]
```diff
@@ -103,7 +103,7 @@
     "@dnd-kit/sortable": "^8.0.0",
     "@dnd-kit/utilities": "^3.2.2",
     "@ibm/telemetry-js": "^1.5.0",
-    "framer-motion": "^6.5.1 < 7",
+    "framer-motion": "11.11.17^",
```
Comment (elycheea) [inline packages/ibm-products/package.json:106]:
```suggestion
    "framer-motion": "^11.11.17",
```

**Comment by matthewgallo** [comment]
@AlexanderMelox `yarn.lock` also needs an update

File: package.json [hunk 2]
```diff
@@ -268,6 +268,7 @@
     ]
   },
   "dependencies": {
+    "framer-motion": "^11.11.17",
```
Comment (elycheea) [inline package.json:271]: Was there a reason we added it to the root `package.json` rather than just updating it as part of our `ibm-products` package?

Comment (AlexanderMelox) [reply package.json:271]: No that was an accident!

**Review by elycheea — APPROVED** [review]
Thank you @AlexanderMelox and @davidmenendez for getting those tests fixed. 🙏

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**